portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -70,4 +116,25 @@ private String getParameter(HttpServletRequest request, String name, String defa
     }
     return value;
   }
+
+  private int getLimit(HttpServletRequest request) {
+    // Get the input from the form.
+    String limitString = request.getParameter("limit-input");
+
+    // Convert the input to an int.
+    int limit;
+    try {
+      limit = Integer.parseInt(limitString);
+    } catch (NumberFormatException e) {
+      System.err.println("Could not convert to int: " + limitString);
+      return -1;
```

Comment:
It's good that you're handling these errors, especially since someone could send a URL with any value in it!

Will returning -1 here cause problems for the `subList()` call that uses the result of this function?

Reply:
I think an if statement here will solve the issue, I will put that in the code as well